### PR TITLE
AutoHCK fail when rerun test with BSOD

### PR DESCRIPTION
**Pull request by Jedoku**
Resolves: #254

### CODE REVIEW

File: lib/engines/hcktest/tests.rb [hunk 1]
```diff
@@ -307,7 +307,7 @@ def collect_memory_dumps(test)
       l_zip_path = "#{@project.workspace_path}/memory_dump_#{id}.zip"
       l_tmp_path = "#{@project.workspace_path}/tmp_#{id}"
 
-      collected_client = collect_memory_dump(@client.name, "#{l_tmp_path}/#{@client.name}")
+      collected_client = collect_memory_dump(@client.name, "#{l_tmp_path}/#{@client.name}_#{timestamp}")
       collected_support = collect_memory_dump(@support.name, "#{l_tmp_path}/#{@support.name}") unless @support.nil?
```
Comment (kostyanf14) [inline lib/engines/hcktest/tests.rb:311]: Both clients can generate BSOD. So add timestamp for support VM too

**Review by kostyanf14 — CHANGES_REQUESTED** [review]

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Can you have an explanation for me, who doesn't know the context? For me, it looks immune against collisions even without a timestamp since the zip has the id.

Why did you use id instead of timestamp? It's inconsistent.
If you really need timestsamps, extract `AutoHCK::Project#create_timestamp` into a helper and change this to use it.

**Comment by Jedoku** [comment]
> Can you have an explanation for me, who doesn't know the context? For me, it looks immune against collisions even without a timestamp since the zip has the id.
> 
> Why did you use id instead of timestamp? It's inconsistent. If you really need timestsamps, extract `AutoHCK::Project#create_timestamp` into a helper and change this to use it.

When we start a test with a dump twice, we have to save both dumps into a zip archive. The problem arises when we save a folder with the name of the client's VM (CL01 in this case) into the zip file. When we try to add the folder a second time, we receive an error stating that the folder already exists. The solution to this issue involves adding a timestamp to the folder name, allowing us to circumvent the problem.

**Comment by akihikodaki** [comment]
@Jedoku I see. Thanks for the explanation.

File: lib/auxiliary/time_helper.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def time_diff(start, finish)
     end
 
     def timestamp
```
Comment (akihikodaki) [inline lib/auxiliary/time_helper.rb:28]: Naming this timestamp will collide with `Project#timestamp` attr_reader so it's better to avoid.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/auxiliary/time_helper.rb [hunk 3]
```diff
@@ -24,5 +24,9 @@ def seconds_to_time(sec)
     def time_diff(start, finish)
       ((finish - start) * 24 * 60 * 60).to_i
     end
+
+    def add_timestamp
```
Comment (kostyanf14) [inline lib/auxiliary/time_helper.rb:28]: This function does not add a timestamp it gets a timestamp. As `get_timestamp` is a bad name from a Ruby point of view, use `create_timestamp`, `generate_timestamp`, `current_timestamp`, etc

**Review by kostyanf14 — CHANGES_REQUESTED** [review]

**Review by kostyanf14 — APPROVED** [review]

File: lib/engines/hcktest/tests.rb [hunk 4]
```diff
@@ -301,16 +301,23 @@ def collect_memory_dump(machine, l_tmp_path)
       true
     end
 
+    def download_memory_dumps(l_tmp_path)
+      collected_client = download_memory_dump(@client.name, "#{l_tmp_path}/#{@client.name}_#{current_timestamp}")
```
Comment (akihikodaki) [inline lib/engines/hcktest/tests.rb:305]: We only download dumps here; please rename `collected_client` to `downloaded_client`. Do the same for `collected_support` too.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

**Review by akihikodaki — APPROVED** [review]

**Pull request merged**